### ISSUE TITLE
Hubspot CRM integration returning successful in CMS, but in Hubspot, API call log is empty and no new contacts are created

### ISSUE THREAD

**Issue by toddpadwick** [verbb/formie#653]
**Description**
I have set up a Hubspot integration with formie forms. The integration says its connected, and when i submit a form, it says integration trigger successful. 
However, in Hubspot, there is nothing in the API call log, and no new contacts being created. 

Is this some kind of bug or have I missed a crucial step here?

**Steps to reproduce**
1. Set up hubspot integration with Hubspot API key
2. Connect with Formie
3. Set up integration with formie form
4. Submit a form or trigger a CRM connection with an already existing submission
5. success message is shown
6. Nothing in hubspot

**Additional info**
- Plugin version: dev-craft-3 as 1.4.25
- Craft version: 3.7.20

**Comment by toddpadwick** [comment]
I've also just discovered, that when saving a form submission in the CMS itself, some required fields clear and become empty after the form is re-saved.  is this due to some mismatch in a two way API connection with Hubspot whereby hubspot is clearing the fields in Formie submissions?

**Comment by engram-design** [comment]
Can you consult your log files to see if there's any indication of information (response back from HubSpot)? Are the queue jobs created once the submission is created failing, or going through okay? You can view the queue via Utilities > Queue Manager.

There should be nothing like as you describe (no two-way connection), that seems odd. What are the field types that clear?

Just so I can confirm:
1. You have some required fields in your Form
2. You do a front-end submission
3. You view/edit the submission in the control panel. All looks fine
4. You edit the form without making any changes
5. You go back to the same submission, and now values have gone for some fields?

**Comment by toddpadwick** [comment]
Hi @engram-design 
<details>
<summary>
I can't fathom much from the logs - see here
</summary>
```
apache_access  91.126.213.35 - - [25/Nov/2021:11:46:54 +0000] "GET /admin/actions/debug/default/toolbar?p=admin/actions/debug/default/toolbar&tag=619f77adf3c6d HTTP/1.1" 404 1215 "https://cms.functionandform.co.uk/admin/formie/submissions" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:46:57 +0000] "GET /admin/actions/debug/default/toolbar?p=admin/actions/debug/default/toolbar&tag=619f77b1688a1 HTTP/1.1" 404 1215 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:08 +0000] "GET /cpresources/684d37fc/images/spinner_details.gif HTTP/1.1" 200 3879 "https://cms.functionandform.co.uk/cpresources/684d37fc/css/craft.css?v=1637766567" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:08 +0000] "POST /index.php?p=admin/actions/formie/submissions/run-integration HTTP/1.1" 200 1306 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:09 +0000] "GET /cpresources/514122ea/css/formie-cp.css?v=1637840762 HTTP/1.1" 200 4788 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:08 +0000] "GET /admin/formie/submissions/projectenquiry/20635?p=admin/formie/submissions/projectenquiry/20635 HTTP/1.1" 200 29378 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:09 +0000] "GET /cpresources/514122ea/js/formie-cp.js?v=1637840762 HTTP/1.1" 200 13097 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:09 +0000] "GET /cpresources/39001481/formie.js?v=1637840787 HTTP/1.1" 200 26207 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:09 +0000] "GET /cpresources/c88cafeb/phone-country.css?v=1637840787 HTTP/1.1" 200 3498 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:09 +0000] "GET /cpresources/d8a4b65d/conditions.js?v=1637840788 HTTP/1.1" 200 16851 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:09 +0000] "GET /cpresources/d8a4b65d/phone-country.js?v=1637840788 HTTP/1.1" 200 23887 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:09 +0000] "GET /admin/actions/debug/default/toolbar?p=admin/actions/debug/default/toolbar&tag=619f77bcc98a1 HTTP/1.1" 404 1215 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:21 +0000] "POST /admin/formie/submissions/projectenquiry/20635?p=admin/formie/submissions/projectenquiry/20635 HTTP/1.1" 302 1336 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:22 +0000] "GET /admin/formie/submissions?p=admin/formie/submissions HTTP/1.1" 200 26984 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:23 +0000] "GET /cpresources/514122ea/css/formie-cp.css?v=1637840762 HTTP/1.1" 200 4788 "https://cms.functionandform.co.uk/admin/formie/submissions" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:23 +0000] "GET /cpresources/514122ea/js/formie-cp.js?v=1637840762 HTTP/1.1" 200 13097 "https://cms.functionandform.co.uk/admin/formie/submissions" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:23 +0000] "POST /index.php?p=admin/actions/queue/get-job-info&limit=50&dontExtendSession=1 HTTP/1.1" 200 927 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:23 +0000] "POST /index.php?p=admin/actions/element-indexes/get-elements&v=1637840843186 HTTP/1.1" 200 7107 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:23 +0000] "POST /index.php?p=admin/actions/element-indexes/count-elements&v=1637840843620 HTTP/1.1" 200 1322 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:23 +0000] "POST /index.php?p=admin/actions/formie/charts/get-submissions-data HTTP/1.1" 200 5116 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:24 +0000] "POST /index.php?p=admin/actions/queue/get-job-info&limit=50&dontExtendSession=1 HTTP/1.1" 200 929 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:24 +0000] "GET /admin/formie/submissions/projectenquiry/20635?p=admin/formie/submissions/projectenquiry/20635 HTTP/1.1" 200 29339 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:25 +0000] "GET /cpresources/d8a4b65d/conditions.js?v=1637840788 HTTP/1.1" 200 16851 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:25 +0000] "GET /cpresources/d8a4b65d/phone-country.js?v=1637840788 HTTP/1.1" 200 23887 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:25 +0000] "GET /cpresources/39001481/formie.js?v=1637840787 HTTP/1.1" 200 26207 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:25 +0000] "GET /cpresources/c88cafeb/phone-country.css?v=1637840787 HTTP/1.1" 200 3498 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:23 +0000] "GET /admin/actions/debug/default/toolbar?p=admin/actions/debug/default/toolbar&tag=619f77cac8f44 HTTP/1.1" 404 1215 "https://cms.functionandform.co.uk/admin/formie/submissions" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
apache_access  91.126.213.35 - - [25/Nov/2021:11:47:25 +0000] "GET /admin/actions/debug/default/toolbar?p=admin/actions/debug/default/toolbar&tag=619f77cd2f310 HTTP/1.1" 404 1215 "https://cms.functionandform.co.uk/admin/formie/submissions/projectenquiry/20635" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
```
</details>

I have now realised the reason why some fields were emptying on submit is because 'Conditions' were enabled (but with no actual conditions added). So although its still buggy, I can get around that now by disabling conditions.

But having done, so Hubspot still is not receiving any data.

- I have tried submitting on the front end and in the submissions dashboard in the CMS
- When i view in the CMS, all looks fine
- Not sure what you mean by editing the form without making changes?
- When returning back to the same submission, some values were emptying however, as above, I determined this is a bug due to conditions being enabled

**Comment by engram-design** [comment]
@toddpadwick Sorry, I should've clarified your Craft/Formie logs in `storage/logs/formie.log`.

Ah, I thought that might've been the case with conditions, I'll check that out shortly. And yep, my steps were just outlining your issue with field values disappearing.

By default, anything going to HubSpot will use Craft's queue system, which I why I recommended just to check if there's any stuck tasks in the queue, but sounds like there isn't.

**Comment by toddpadwick** [comment]
@engram-design  Ah, frustratingly as we're using Fortrabbit Pro stack, we can only tail global apache/php logs rather than see historic logs in the storage folder. https://help.fortrabbit.com/logging-pro

But i tried on my local instance of the site and that didnt work either. However, there is no storage/logs/formie.log file generated even on my local. Is there a way to enable formie logs?

**Comment by engram-design** [comment]
Ah, no problem. Logs are always enabled, but that's strange **nothing** appears when you make a form submission? It'll log every ping to the `formie/submissions/submit` endpoint, so there should always be something in that file.

Are you able to share your Integration mapping settings (edit the form, Integrations tab, highlight HubSpot)?

If you go to a successful submission, and re-trigger the integration (the new-ish button on the sidebar): 
![image](https://user-images.githubusercontent.com/1221575/143503169-932f2541-30f8-41f6-83b8-0f1210969ef7.png)

What happens in that case?

**Comment by toddpadwick** [comment]
Hi @engram-design 
I was already using the integrations dropdown button in the previous tests. 

There seems to be no formie log file in both our production server and localhost :( 

Here are my CRM mapping settings:

![Screenshot 2021-11-26 at 10 56 15](https://user-images.githubusercontent.com/36191462/143562450-c35ac37f-2d9b-456c-a517-80f181861303.png)


![Screenshot 2021-11-26 at 10 56 19](https://user-images.githubusercontent.com/36191462/143562468-b2cf3306-c5d4-4575-9ee2-9ff5ab3ac88f.png)

**Comment by engram-design** [comment]
Thanks for that, that all certainly looks valid. I can't tell if you have the HubSpot integration enabled in the sidebar, but I assume you do.

When you run the integration from the submission again, does it succeed and show `Integration was run successfully?`

Slightly at a bit of a loss, short of adding some debugging statements in the code:

Try adding a `Craft::dd('STOP');` debug statement at https://github.com/verbb/formie/blob/craft-3/src/integrations/crm/HubSpot.php#L240 (right above the `try` statement), then try running the integration from the submission again. It should show `A server error occurred.`

**Comment by toddpadwick** [comment]
Hi @engram-design 
I tried adding the stop function. It returned, as expected the 'server error occurred' message.
Still no log file generated. (on localhost)

**Comment by toddpadwick** [comment]
I have just run a test on an alternative CRM integration - copper - and I can confirm this does not work either.

**Comment by engram-design** [comment]
So it's certainly reaching the HubSpot integration class.

In Formie > Settings > Submissions, what is the setting for "Use Queue for Integrations"? If this is on, can you confirm that there are no queue jobs stuck in Utilities > Queue Manager?

**Comment by toddpadwick** [comment]
We have our queue manager running as a non stop worker in Fortrabbit. This seems to be working fine on other native craft jobs. I cannot, however, see any formie jobs in the queue or any in the logs when i tail the worker jobs. I have messaged Fortrabbit for support as well to see if they can shed any further light on this as I dont really understand how the queue works or how their worker system works.

**Comment by engram-design** [comment]
No problem, it sounds like there's some sort of queue worker going - which is good - as it probably isn't that. I can't say I'm familiar with Fortrabbit's logging, but it's **very** strange there's no `storage/logs/formie.log` file, even on local - that doesn't quite make sense. Are there _any_ logs in that folder?

Try adding the following:

```
Integration::log($this, 'mapToContact: ' . $this->mapToContact);
Integration::log($this, Json::encode($contactValues));
```

At https://github.com/verbb/formie/blob/craft-3/src/integrations/crm/HubSpot.php#L247

**Comment by toddpadwick** [comment]
@engram-design  okay we seem to be getting somewhere as those to loggers have created the formie.log file. Does that shed any further light on the source of the problem?
```
2021-11-30 14:12:41 [-][1][-][info][formie] Hubspot: mapToContact: 
2021-11-30 14:12:41 [-][1][-][info][formie] Hubspot: []
2021-11-30 14:12:41 [-][1][-][info][application] $_GET = [
    'p' => 'admin/actions/formie/submissions/run-integration'
]

$_POST = [
    'submissionId' => '6534'
    'integrationId' => '1'
]
```

To answer your other question, yes there are all the other normal log files as expected: - _phperrors.log_, _console.log_, _queue.log_, _web.log_, _web-404.log_

**Comment by toddpadwick** [comment]
@engram-design I spoke with fortrabbit, to rule out any issues with the worker or queue - they've dug into the log files on disk and can confirm no "verbb\formie\jobs\*" job were handled/sent to the worker from formie.

**Comment by toddpadwick** [comment]
Fortrabbit guys informed me that this request returned a 200 OK
`POST https://cms.functionandform.co.uk/index.php?p=admin/actions/formie/submissions/run-integration`
and suggested that that means it gets to this point in the code okay:
https://github.com/verbb/formie/blob/craft-3/src/controllers/SubmissionsController.php#L875
So hopefully that narrows things down a little more.

**Comment by engram-design** [comment]
So that's interesting, the log line:
```
2021-11-30 14:12:41 [-][1][-][info][formie] Hubspot: mapToContact: 
```

Shows that `$this->mapToContact` is false (otherwise it'd show `mapToContact: true`). Can you confirm if you're mapping to the contact? I can see in your screenshot you have that lightswitch turned on, but just wanted to double check. Even refreshing and re-saving the form (make sure you're editing the HubSpot integration) just to be totally sure.

I'm thinking that if it's not finding the Contact/Deal lightswitches enabled then it won't do anything at all, which seems like what's happening here (no logs, no nothing), because nothing is enabled.

If you can share your PHP version, and also the row in your `formie_forms` database table that is for the form you're editing. I'm interested in the integration settings for the form, and if they're not saving these values correctly. If it's affecting other integrations, it might be a PHP issue, or just something wrong with how the form integration settings are saved.

**Comment by toddpadwick** [comment]
@engram-design  I can confirm map to contact and map to deal are enabled. I've refreshed and re-saved as well to confirm. i also have email field, name fields and a number of other fields mapped.
I've emailed the DB over to the email you have in your github profile. The form thats integrated is _projectEnquiry_.

The PHP version is 8.0

**Comment by engram-design** [comment]
Thanks for the database, that's help immensely. I've identified a few issues, one of which I think might be largely the issue. It's where the integration when re-triggered from the submission wasn't being properly prepped with the correct settings. Hence, it was showing none of your objects like contact/deal were being opted-into.

Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.27",
  "...": "..."
}
```

Then run `composer update`.

**Comment by toddpadwick** [comment]
Hi @engram-design  I can confirm this works now for Hubspot. But it doesnt seem to work for Copper. Should this fix have solved all integrations or just Hubspot?

**Comment by toddpadwick** [comment]
@engram-design Also, on a side note, you may not be aware Copper have changed their API endpoint. But as they are re-directing until early 2022, it shouldn't be the cause of the issue. - but probably worth changing all the same. https://developer.copper.com/?_ga=2.39736739.1218310121.1638367599-498473556.1638198625

**Comment by engram-design** [comment]
Great to hear about HubSpot, but the fix should've been integration-wide. Good call on the API change though - easy fix!

Are you able to try the above command again (run `composer update`) and see if that helps?.

**Comment by toddpadwick** [comment]
Hey @engram-design getting an internal server error now:
`Request must accept JSON in response`

<details>
<summary>
 See stack trace
</summary>
```
yii\web\BadRequestHttpException: Request must accept JSON in response in /srv/app/ffd-cms/htdocs/vendor/craftcms/cms/src/web/Controller.php:329
Stack trace:
#0 /srv/app/ffd-cms/htdocs/vendor/verbb/formie/src/controllers/SubmissionsController.php(848): craft\web\Controller->requireAcceptsJson()
#1 [internal function]: verbb\formie\controllers\SubmissionsController->actionRunIntegration()
#2 /srv/app/ffd-cms/htdocs/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#3 /srv/app/ffd-cms/htdocs/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams(Array)
#4 /srv/app/ffd-cms/htdocs/vendor/yiisoft/yii2/base/Module.php(534): yii\base\Controller->runAction('run-integration', Array)
#5 /srv/app/ffd-cms/htdocs/vendor/craftcms/cms/src/web/Application.php(287): yii\base\Module->runAction('formie/submissi...', Array)
#6 /srv/app/ffd-cms/htdocs/vendor/craftcms/cms/src/web/Application.php(596): craft\web\Application->runAction('formie/submissi...', Array)
#7 /srv/app/ffd-cms/htdocs/vendor/craftcms/cms/src/web/Application.php(266): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#8 /srv/app/ffd-cms/htdocs/vendor/yiisoft/yii2/base/Application.php(392): craft\web\Application->handleRequest(Object(craft\web\Request))
#9 /srv/app/ffd-cms/htdocs/web/index.php(21): yii\base\Application->run()
#10 {main}
```
</details>

**Comment by engram-design** [comment]
Strange, it doesn't seem to treat the run-integration endpoint as being run via Ajax, which it should be. Are there any JS error when you load the edit submission page?

Is this running HubSpot, Copper or both?

**Comment by toddpadwick** [comment]
This is running just copper using the 'run integration' dropdown
Here are two errors. one (the reCaptcha error) maybe related to this issue which [my developer just posted](https://github.com/verbb/formie/issues/669)?

<img width="524" alt="Screenshot 2021-12-02 at 12 07 20" src="https://user-images.githubusercontent.com/36191462/144410878-3208919b-073f-4434-b218-994755d33664.png">

**Comment by engram-design** [comment]
If you inspect the latter error in your Network tab, and click on the Preview in that, what do you see?

![image](https://user-images.githubusercontent.com/1221575/144415615-dd8433a9-8a46-47d1-b6a8-f731c7d8e8d2.png)

I've also found an issue with Copper that failed matching an existing person. Their API endpoint for https://developer.copper.com/people/fetch-a-person-by-email.html just fails, so I've had to switch it to something else (`composer update` again).

I don't _think_ the recaptcha error should cause an issue here, but I've also fixed that.

**Comment by toddpadwick** [comment]
@engram-design Heres the preview:

<img width="775" alt="Screenshot 2021-12-02 at 13 41 44" src="https://user-images.githubusercontent.com/36191462/144424102-f9e78d7b-bc4a-41a9-a9da-3e09d6b01b9f.png">

```
error: "API error: “{\"success\":false,\"status\":404,\"message\":\"Resource not found\"}” /srv/app/ffd-cms/htdocs/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113"
exception: "verbb\\formie\\errors\\IntegrationException"
file: "/srv/app/ffd-cms/htdocs/vendor/verbb/formie/src/base/Integration.php"
line: 166
trace: [{file: "/srv/app/ffd-cms/htdocs/vendor/verbb/formie/src/integrations/crm/Copper.php", line: 612,…},…]
```

**Comment by engram-design** [comment]
Yep, that `Resource not found` issue was the one I mentioned above for the API endpoint seemingly not being able to be found. The latest updates should fix that.

**Comment by toddpadwick** [comment]
Okay, so that fixed the resource not found issue. But I then got two more errors... First, a validation error: 
```
error: "API error: “{\"success\":false,\"status\":422,\"message\":\"Invalid input: Validation errors: Company Id: Invalid value for Company Id: IDs must be positive integers: test\"}” /srv/app/ffd-cms/htdocs/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113"
exception: "verbb\\formie\\errors\\IntegrationException"
file: "/srv/app/ffd-cms/htdocs/vendor/verbb/formie/src/base/Integration.php"
line: 166
trace: [{file: "/srv/app/ffd-cms/htdocs/vendor/verbb/formie/src/integrations/crm/Copper.php", line: 613,…},…]
```

I believe this is due to the field not matching the field type matched in Copper which is my fault - i have rectified this now. but it would be good to have the error shown somewhere easier to see then in the network tab. I imagine thats going to be hard to debug for some people.

But now, after I fixed that, I now get this error:
```
error: "API error: “{\"success\":false,\"status\":500,\"message\":\"Server error\"}” /srv/app/ffd-cms/htdocs/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113"
exception: "verbb\\formie\\errors\\IntegrationException"
file: "/srv/app/ffd-cms/htdocs/vendor/verbb/formie/src/base/Integration.php"
line: 166
trace: [{file: "/srv/app/ffd-cms/htdocs/vendor/verbb/formie/src/integrations/crm/Copper.php", line: 613,…},…]

```

**Comment by engram-design** [comment]
Yep, I'll improve the feedback for this, similar to the "connect/refresh" modal when testing an integration in settings.

Well that's a frustrating error, but it is coming from Copper's API. Very difficult to debug without un-mapping individual values, but it's quite possible we're not preparing values for some data types in Copper.

**Comment by toddpadwick** [comment]
Is that something I need to fix on my end with regards to mapping fields or is it an issue with the integration still?

**Comment by engram-design** [comment]
It's an issue with the integration still, but I think it might be down to the specific fields you're mapping to Copper fields. I would do a trial by elimination un-mapping things - or feel free to shoot through your updated database with your Copper fields mapped, so I can test if I can replicate it on my end.

**Comment by engram-design** [comment]
Fixed in [1.4.28](https://github.com/verbb/formie/releases/tag/1.4.28)
